### PR TITLE
gui: Volume slider for feedback sound

### PR DESCRIPTION
**Pull request by 0forks**
Definitely missing a volume slider.

I found mids on distortion at the start quite overbearing, but maybe this is just a me issue.

image

### CODE REVIEW

File: gui/src/components/settings/pages/GeneralSettings.tsx [hunk 1]
```diff
@@ -876,6 +884,19 @@ export function GeneralSettings() {
...
             />
           </div>
+          <div className="grid sm:grid-cols-2 pb-4">
+            <NumberSelector
+              control={control}
+              name="interface.feedbackSoundVolume"
+              label={l10n.getString(
+                'settings-general-interface-feedback_sound-volume'
+              )}
+              valueLabelFormat={(value) => `${Math.round(value * 100)} %`}
```
Comment (ImUrX) [inline gui/src/components/settings/pages/GeneralSettings.tsx:894]: Can you use ``Intl.NumberFormat`` with  percentage mode? I use it in the Manual Proportions settings

**Review by ImUrX — CHANGES_REQUESTED** [review]

File: gui/src/components/settings/pages/GeneralSettings.tsx [hunk 2]
```diff
@@ -116,7 +117,12 @@ const defaultValues = {
     mountingResetTaps: 3,
   },
   legTweaks: { correctionStrength: 0.3 },
-  interface: { devmode: false, watchNewDevices: true, feedbackSound: true },
+  interface: {
+    devmode: false,
+    watchNewDevices: true,
+    feedbackSound: true,
+    feedbackSoundVolume: 0.8,
```
Comment (Louka3000) [inline gui/src/components/settings/pages/GeneralSettings.tsx:124]: Honestly it breaks my ears from how loud it was/is by default.
```suggestion
    feedbackSoundVolume: 0.5,
```

**Review by Louka3000 — CHANGES_REQUESTED** [review]

**Review by Louka3000 — APPROVED** [review]

**Review by ImUrX — APPROVED** [review]
space is dependant of language but I dont honestly care XD

**Comment by 0forks** [comment]
IMO without the space it blends with the numbers too much, making it hard to read.

**Comment by 0forks** [comment]
Okay, I hate it, but I hate UI inconsistencies more.

File: gui/src/components/settings/pages/GeneralSettings.tsx [hunk 3]
```diff
@@ -116,15 +118,26 @@ const defaultValues = {
     mountingResetTaps: 3,
   },
   legTweaks: { correctionStrength: 0.3 },
-  interface: { devmode: false, watchNewDevices: true, feedbackSound: true },
+  interface: {
+    devmode: false,
+    watchNewDevices: true,
+    feedbackSound: true,
+    feedbackSoundVolume: 0.5,
```
Comment (Spazznyan) [inline gui/src/components/settings/pages/GeneralSettings.tsx:125]: Just voting for a tad lower for the default, better to have it too low than too high imo. Im just assuming 0.5 is linear 50% and not logarithmic? Something like 0.2 would be preferable for myself as most of the time 50% is too high in my setup.

Comment (0forks) [reply gui/src/components/settings/pages/GeneralSettings.tsx:125]: I made it logarithmic, 50% will set it to ~0.25 (in whatever HTML audio units), which peaks at -16 dBFS and 20% is -24 dBFS. I believe 50% is a sane default.

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**